dev/com.ibm.ws.concurrent.mp.cdi/src/com/ibm/ws/concurrent/mp/cdi/MPConfigAccessorImpl.java

```diff
@@ -0,0 +1,46 @@
+/*******************************************************************************
+ * Copyright (c) 2018 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.concurrent.mp.cdi;
+
+import java.util.Optional;
+
+import org.eclipse.microprofile.config.Config;
+import org.eclipse.microprofile.config.spi.ConfigProviderResolver;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * This class abstracts away usage of MicroProfile Config so that values can be
+ * conditionally read from MicroProfile Config if available. The only data types
+ * needed are String[] and Integer.
+ */
+@Component(configurationPolicy = ConfigurationPolicy.IGNORE)
+public class MPConfigAccessorImpl implements MPConfigAccessor {
+    @Reference
+    protected ConfigProviderResolver configProviderResolver;
+
+    /**
+     * Reads a String[] or Integer property value from MicroProfile Config.
+     *
+     * @param name config property name.
+     * @param defaultValue value to use if a config property with the specified name is not found.
+     * @return value from MicroProfile Config. Otherwise the default value.
+     */
+    @Override
+    public <T> T get(String name, T defaultValue) {
+        Config config = configProviderResolver.getConfig();
```

Comment:
Looking through our implementation of ConfigProviderResolver, this seems like a rather expensive operation with several synchronized blocks in it.  Lets lazily initialize and cache the `Config` instance per `MPConfigAccessorImpl` instance

Reply:
We will look into improving this path under subsequent pulls, however, it isn't as simple as caching a single config instance for the MPConfigAccessorImpl singleton because that singleton is used to load config that can differ by application due to microprofile-config.properties.  Before we do anything in this area, I would like to have added a test that utilizes MP config per application to ensure we aren't breaking it.